packages/jellyfish-api-core/__tests__/category/wallet/getBalances.test.ts

```diff
@@ -0,0 +1,89 @@
+import { MasterNodeRegTestContainer, RegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { BigNumber } from '../../../src'
+import { WalletFlag, WalletBalances } from '../../../src/category/wallet'
+
+// TODO(aikchun): Add behavior tests for untrusted_pending, immature, used. Currently unable to do multi-node testing
+describe('getBalances on masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should getBalances', async () => {
+    const balances: WalletBalances = await client.wallet.getBalances()
+    expect(balances.mine.trusted instanceof BigNumber).toStrictEqual(true)
+    expect(balances.mine.untrusted_pending instanceof BigNumber).toStrictEqual(true)
+    expect(balances.mine.immature instanceof BigNumber).toStrictEqual(true)
+    expect(typeof balances.mine.used).toStrictEqual('undefined')
+
+    expect(typeof balances.watchonly).toStrictEqual('undefined')
+  })
+
+  it('should have used in getBalances when wallet is set to avoid_reuse', async () => {
```

Comment:
what's the diff after setting avoid_reuse (true/false) on getBalances??

Reply:
This is to set the `walletFlag` to `avoid_reuse`. when a wallet is set to `avoid_reuse`, the `mine.used` property will be set. Otherwise, `mine.used` is `undefined`.